minimal-plugin/src/main/java/io/micronaut/gradle/MicronautRuntime.java

```diff
@@ -22,98 +16,69 @@
-            Collections.singletonList("io.micronaut.oraclecloud:micronaut-oraclecloud-function-http-test"),
-            JavaPlugin.RUNTIME_ONLY_CONFIGURATION_NAME,
-            Collections.singletonList("com.fnproject.fn:runtime")
-    )),
+    ORACLE_FUNCTION(DockerBuildStrategy.ORACLE_FUNCTION),
     /**
      * Google Cloud Function, packaged as a Fat JAR.
      */
-    GOOGLE_FUNCTION(MicronautExtension.mapOf(
-            JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME,
-            Collections.singletonList("io.micronaut.gcp:micronaut-gcp-function-http"),
-            "developmentOnly",
-            Arrays.asList("com.google.cloud.functions:functions-framework-api", "io.micronaut.gcp:micronaut-gcp-function-http-test"),
-            JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME,
-            Collections.singletonList("com.google.cloud.functions:functions-framework-api"),
-            JavaPlugin.TEST_IMPLEMENTATION_CONFIGURATION_NAME,
-            Arrays.asList("com.google.cloud.functions:functions-framework-api", "io.micronaut.gcp:micronaut-gcp-function-http-test")
-    )),
+    GOOGLE_FUNCTION,
     /**
      * Azure Cloud Function.
      */
-    AZURE_FUNCTION(MicronautExtension.mapOf(
-            JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME,
-            Arrays.asList("io.micronaut.azure:micronaut-azure-function-http", "com.microsoft.azure.functions:azure-functions-java-library"),
-            "developmentOnly",
-            Collections.singletonList("io.micronaut.azure:micronaut-azure-function-http-test"),
-            JavaPlugin.TEST_IMPLEMENTATION_CONFIGURATION_NAME,
-            Collections.singletonList("io.micronaut.azure:micronaut-azure-function-http-test")
-    ));
+    AZURE_FUNCTION;
 
-    private final Map<String, List<String>> implementation;
     private final DockerBuildStrategy buildStrategy;
 
-    MicronautRuntime(String... dependencies) {
-        this.implementation = Collections.singletonMap(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME, Arrays.asList(dependencies));
+    MicronautRuntime() {
         this.buildStrategy = DockerBuildStrategy.DEFAULT;
     }
 
-    MicronautRuntime(Map<String, List<String>> implementation) {
-        this.implementation = implementation;
-        this.buildStrategy = DockerBuildStrategy.DEFAULT;
-    }
-
-    MicronautRuntime(DockerBuildStrategy buildStrategy, Map<String, List<String>> implementation) {
-        this.implementation = implementation;
+    MicronautRuntime(DockerBuildStrategy buildStrategy) {
         this.buildStrategy = buildStrategy;
     }
 
-    /**
-     * A map of dependencies and scopes
-     * @return The dependencies and scopes
-     */
-    public Map<String, List<String>> getDependencies() {
-        return implementation;
-    }
-
     /**
      * @return The docker build strategy
      */
     public DockerBuildStrategy getBuildStrategy() {
         return buildStrategy;
     }
-}
+
+    public static boolean isLambdaProvided(MicronautRuntime runtime) {
```

Comment:
:o: Why not having an instance method here? `mr.isLambdaProvided()`?

Reply:
https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/434/commits/12f3d78b8a09b02732709d5e42c71bfbe68b7572